tcblib/slack_functions.py

```diff
@@ -0,0 +1,52 @@
+import boto3
+import hmac
+import slack
+
+def validate_request(req):
+    """Validate slack request to prevent the hax."""
+    slack_signing_secret = boto3.client('ssm').get_parameter(
+        Name='/slack/tcb/signing_secret',
+        WithDecryption=True
+    )['Parameter']['Value']
+    signing_secret = str.encode(slack_signing_secret)
+
+    slack_request_timestamp = req['headers']['X-Slack-Request-Timestamp']
+    request_body = req['body']
+
+    token = 'v0:{0}:{1}'.format(
+        slack_request_timestamp,
+        request_body
+    )
+    encoded_token = str.encode(token)
+
+    slack_signature = req['headers']['X-Slack-Signature']
```

Comment:
Just as a general rule of thumb I'd either put all of these helper/convenience variables up at the top near each other, or right next to its actual use.